assets/js/common.js

```diff
@@ -554,6 +564,41 @@ var hivepress = {
 								fields.eq(0).val(formattedDates[0]);
 								fields.eq(1).val(formattedDates[1]);
 							} else {
+								var days = instance.calendarContainer.querySelectorAll('.flatpickr-day'),
```

Comment:
There's an issue with this way of selecting days, it selects days within the current month view only. I tried selecting the start date at the end of the month and the end date next month, passing the End Date Only date and it allowed to do this. Please check the Flatpickr docs maybe they have a method for getting past/future dates as objects, it's better to avoid querying days by CSS class.

Reply:
P.S. Switch to the "release" branch in Bookings when testing this, I moved the "last-date" change to a new branch.

Reply:
If there's 100% no suitable method for fetching dates dynamically beyond the current month then a possible solution is collecting a plain JS array of dates first or just dates before/after which we should block dates (based on the latest previous End Date Only and the first next End Date Only, so 2 variables may be ok), and maybe there's an event callback which fires when the month is switched - then we can go through the dates again and grey them out if they are < or > of these 2 possible limits.